From 8695cf263f983b55388aff7d7642b92e3c5f9173 Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Thu, 27 Oct 2016 14:23:20 -0500
Subject: [PATCH 2075/2848] HDFS-11069. Tighten the authorization of datanode
 RPC. Contributed by Kihwal Lee Updated
 CHANGES.txt (cherry picked from commit
 ae48c496dce8d0eae4571fc64e6850d602bae688)

(cherry picked from commit a49510f697c7f64861e4429c3b8bd23b062e6905)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: If96cf9c8b356dcd88c8c570a8965769a5a8bf1a4
---
 .../hadoop/hdfs/server/datanode/DataNode.java      |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java
index 6ef2987..a21ebaf 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java
@@ -935,7 +935,7 @@ private void checkSuperuserPrivilege() throws IOException, AccessControlExceptio
 
     // Is this by the DN user itself?
     assert dnUserName != null;
-    if (callerUgi.getShortUserName().equals(dnUserName)) {
+    if (callerUgi.getUserName().equals(dnUserName)) {
       return;
     }
 
@@ -1271,7 +1271,7 @@ void startDataNode(Configuration conf,
     this.blockPoolTokenSecretManager = new BlockPoolTokenSecretManager();
 
     // Login is done by now. Set the DN user name.
-    dnUserName = UserGroupInformation.getCurrentUser().getShortUserName();
+    dnUserName = UserGroupInformation.getCurrentUser().getUserName();
     LOG.info("dnUserName = " + dnUserName);
     LOG.info("supergroup = " + supergroup);
     initIpcServer(conf);
@@ -3224,4 +3224,4 @@ public String getDiskBalancerSetting(String key) throws IOException {
   void setBlockScanner(BlockScanner blockScanner) {
     this.blockScanner = blockScanner;
   }
-}
\ No newline at end of file
+}
-- 
1.7.9.5

